destination/writer/writer.go

```diff
@@ -0,0 +1,293 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/conduitio-labs/conduit-connector-clickhouse/columntypes"
+	"github.com/conduitio-labs/conduit-connector-clickhouse/repository"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+const (
+	// metadata key with the name of the table.
+	metadataTable = "clickhouse.table"
+
+	// query patterns.
+	insertFmt = "INSERT INTO %s (%s) VALUES (%s)"
+	updateFmt = "ALTER TABLE %s UPDATE %s WHERE %s"
+	deleteFmt = "ALTER TABLE %s DELETE WHERE %s"
+
+	coma        = ","
+	placeholder = "?"
+	operatorAnd = " AND "
+)
+
+// Writer implements a writer logic for ClickHouse destination.
+type Writer struct {
+	repo           *repository.ClickHouse
+	table          string
+	primaryColumns []string
+	columnTypes    map[string]string
+}
+
+// Params is an incoming params for the New function.
+type Params struct {
+	Repo           *repository.ClickHouse
+	Table          string
+	PrimaryColumns []string
+}
+
+// New creates new instance of the Writer.
+func New(ctx context.Context, params Params) (*Writer, error) {
+	writer := &Writer{
+		repo:           params.Repo,
+		table:          params.Table,
+		primaryColumns: params.PrimaryColumns,
+	}
+
+	columnTypes, err := columntypes.GetColumnTypes(ctx, writer.repo, writer.table)
+	if err != nil {
+		return nil, fmt.Errorf("get column types: %w", err)
+	}
+
+	writer.columnTypes = columnTypes
+
+	return writer, nil
+}
+
+// Write writes a sdk.Record into a Destination.
+func (w *Writer) Write(ctx context.Context, record sdk.Record) error {
+	switch record.Operation {
```

Comment:
Let's make use of the `sdk.Util.Destination.Route` method instead of the switch.

Reply:
Good point. Thank you 👍